Common/Orders/Fills/EquityFillModel.cs

```diff
@@ -600,6 +610,35 @@ private decimal GetBidPrice(Security asset, out DateTime endTime)
             throw new InvalidOperationException($"Cannot get bid price to perform fill for {asset.Symbol} because no market data subscription were found.");
         }
 
+        /// <summary>
+        /// Get current trade bar for subscribed data
+        /// This method will try to get the most recent trade data, so it will try to get tick trade first, then trade bar.
+        /// </summary>
+        /// <param name="asset">Security which has subscribed data types</param>
+        private TradeBar GetLastTradeBar(Security asset)
+        {
+            var subscribedTypes = GetSubscribedTypes(asset);
+
+            if (subscribedTypes.Contains(typeof(Tick)))
+            {
+                var trade = asset.Cache.GetAll<Tick>().LastOrDefault(x => x.TickType == TickType.Trade && x.Price > 0);
+                if (trade != null)
+                {
+                    return new TradeBar(trade.Time, trade.Symbol, trade.Price, trade.Price, trade.Price, trade.Price, trade.Quantity, TimeSpan.Zero);
+                }
+            }
+
+            if (subscribedTypes.Contains(typeof(TradeBar)))
+            {
+                var tradeBar = asset.Cache.GetData<TradeBar>();
+                if (tradeBar != null)
+                {
+                    return tradeBar;
+                }
+            }
+
+            throw new InvalidOperationException($"Cannot perform fill for {asset.Symbol} because no trade data were found.");
```

Comment:
Not 100% sure about throwing here, what if no trade has happened yet? as long as there is a subscription to TradeBar or Tick, don't think it should throw?